pkg/apply/apply.go

```diff
@@ -74,9 +74,49 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 			}
 
 			return errors.Wrapf(err, "could not update object %s", objDesc)
-		} else {
-			log.Print("update was successful")
 		}
+		log.Print("update was successful")
+	}
+
+	return nil
+}
+
+// DeleteObject deletes an object in the apiserver
+func DeleteObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstructured) error {
+	name := obj.GetName()
+	namespace := obj.GetNamespace()
+	if name == "" {
+		return errors.Errorf("object %s has no name", obj.GroupVersionKind().String())
+	}
+	gvk := obj.GroupVersionKind()
+	// used for logging and errors
+	objDesc := fmt.Sprintf("(%s) %s/%s", gvk.String(), namespace, name)
+	log.Printf("reconciling %s", objDesc)
+
+	if err := IsObjectSupported(obj); err != nil {
+		return errors.Wrapf(err, "object %s unsupported", objDesc)
+	}
+
+	// Get existing
+	existing := &uns.Unstructured{}
+	existing.SetGroupVersionKind(gvk)
+	err := client.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, existing)
+
+	log.Printf("DBGDBG at delete %v A1", name)
+	if name != "cluster-network-addons" {
```

Comment:
here there is a nit, 
on the last component in the ns , the ns should be deleted if we want to handle it
or we know that one will always exists

Reply:
as Petr raised, the NS also has the cnao pod so it shouldnt be deleted even if no components left, so this bullet is fine
thanks

Reply:
This check should be done on a higher level. This method should be universal (not CNAO specific). It should merely Delete anything given to it.

Reply:
removing it as part of next commit totally, filtering the list of objToRemove so wont need it